### PR TITLE
cli function args parse error improvement Issue3140

### PR DESCRIPTION
**Pull request by JerryKwan**
Improve the error generated when parsing args

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Current implement will output errors such as "unexpected end of input" or "remaining input", which is not meaningful

Issue Number: https://github.com/starcoinorg/starcoin/issues/3140

## What is the new behavior?

The new version will output messages such as "error occurred when decoding 4th paramater , the type should be U128" or ""error occurred when decoding 3rd paramater , the type should be Bool

-
-
-

### CODE REVIEW

**Comment by nkysg** [comment]
@JerryKwan Cargo.lock has changed, but not commit。Please try again。

File: abi/decoder/src/lib.rs [hunk 1]
```diff
@@ -139,11 +139,21 @@ pub fn decode_script(state: &dyn StateView, s: &Script) -> Result<DecodedScript>
             arg_abis
         }
     };
-    let args = arg_abis
-        .iter()
-        .zip(s.args())
-        .map(|(ty, arg)| decode_move_value(ty.type_abi(), arg))
-        .collect::<Result<Vec<_>, _>>()?;
+    let mut args = Vec::<DecodedMoveValue>::new();
```
Comment (nkysg) [inline abi/decoder/src/lib.rs:142]: Vec::<DecodedMoveValue>::new() 可以写成Vec::new(),编译器能推导

File: abi/decoder/src/lib.rs [hunk 2]
```diff
@@ -201,11 +211,20 @@ fn decode_script_function_inner(
             arg_abis
         }
     };
-    let args = arg_abis
-        .iter()
-        .zip(sf.args())
-        .map(|(abi, arg)| decode_move_value(abi.type_abi(), arg))
-        .collect::<Result<Vec<_>, _>>()?;
+    let mut args = Vec::<DecodedMoveValue>::new();
```
Comment (nkysg) [inline abi/decoder/src/lib.rs:214]: Vec::::new() 可以写成Vec::new(),编译器能推导

Comment (JerryKwan) [reply abi/decoder/src/lib.rs:214]: Yes, Rust can infer the type automatically

**Comment by JerryKwan** [comment]
@nkysg Changes are mad accordingly, please take some time to review, thank you

**Comment by nkysg** [comment]
LGTM

**Review by nkysg — APPROVED** [review]

**Pull request merged**